stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/factory/QuotedAttachmentFactory.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.ui.attachments.factory
+
+import androidx.compose.runtime.Composable
+import io.getstream.chat.android.compose.ui.attachments.AttachmentFactory
+import io.getstream.chat.android.compose.ui.attachments.content.FileAttachmentQuotedContent
+import io.getstream.chat.android.compose.ui.attachments.content.ImageAttachmentQuotedContent
+import io.getstream.chat.android.compose.ui.util.hasLink
+import io.getstream.chat.android.compose.ui.util.isFile
+import io.getstream.chat.android.compose.ui.util.isMedia
+
+/**
+ * An [AttachmentFactory] that validates attachments as files and uses [ImageAttachmentQuotedContent] or
+ * [FileAttachmentQuotedContent] to build the UI for the quoted message.
+ */
+@Suppress("FunctionName")
+public fun QuotedAttachmentFactory(): AttachmentFactory = AttachmentFactory(
+    canHandle = { it.isNotEmpty() },
+    content = content@ @Composable { modifier, attachmentState ->
```

Comment:
Since this won't be called if we have an empty list of attachments, we can avoid this label and the `firstOrNull` call can be `first` to avoid early return if null.